### PR TITLE
Track 105 - Enabling multiple clients to grade the same lesson

### PR DESCRIPTION
**Pull request by tomca32**
We are making Lesson creation idempotent.

The reason for this is that when there are multiple clients grading at the same time, they both need to create a lesson before pushing grades. Since we are currently erroring with 422 when trying to create a duplicate lesson, the second client loses and cannot push their grades since it doesn't know with which lesson should the grades be associated. This should solve that issue.

Basically, this PR enables multiple teachers in the same lesson.

### CODE REVIEW

File: app/controllers/api/lessons_controller.rb [hunk 1]
```diff
@@ -16,9 +16,26 @@ def show
     end
 
     def create
-      @lesson = Lesson.new params.permit :group_id, :date, :subject_id
-      @lesson.save
-      respond_with @lesson, include: included_params, meta: { timestamp: Time.zone.now }
+      lesson = Lesson.find_by group_id: params[:group_id], subject_id: params[:subject_id], date: params[:date]
```
Comment (j16r) [inline app/controllers/api/lessons_controller.rb]: This is unfortunately a race condition, you really want to leverage the database to get this right - e.g: constraint violation.

Comment (tomca32) [inline app/controllers/api/lessons_controller.rb]: Huh? Me stupid...don't understand. find_by is a race condition?

**Comment by tomca32** [comment]
@j16r Updated. Put a uniqueness index on the DB and leveraging that.
